reanalysis/comparison.py

```diff
@@ -0,0 +1,154 @@
+
+from argparse import ArgumentParser
+import hail as hl
+
+from reanalysis.utils import read_json_from_path
+
+
+SAMPLE_NUM_RE = re.compile(r'sample_[0-9]+')
+SAMPLE_ALT_TEMPLATE = 'num_alt_alleles_{}'
+
+
+class Confidence(Enum):
+    """
+    enumeration for possible confidence values
+    from AIP we get CERTAIN, as these are found
+    from seqr dump, we use tags to determine confidence
+        - where confidence means 'how confident are we that
+            AIP should be capturing this result'
+    """
+
+    EXPECTED = 'AIP training: Expected'
+    POSSIBLE = 'AIP training: Possible'
+    UNLIKELY = 'AIP training: Unlikely'
+
+    def __lt__(self, other):
+        return self.value < other.value
+
+
+class CommonFormatResult:
+    """
+    a common representation of variant details
+    """
+
+    def __init__(
+        self, chrom: str, pos: int, ref: str, alt: str, confidence: List[Confidence]
+    ):
+        """
+
+        :param chrom:
+        :param pos:
+        :param ref:
+        :param alt:
+        :param confidence:
+        """
+        self.chr: str = chrom
+        self.pos: int = pos
+        self.ref: str = ref
+        self.alt: str = alt
+        self.confidence: List[Confidence] = confidence
+
+    def get_cyvcf2_pos(self):
+        """
+        get variant coordinates in string format
+        :return:
+        """
+        return f'{self.chr}:{self.pos}-{self.pos}'
+
+    def get_hail_pos(self) -> Tuple[hl.Locus, List[str]]:
+        """
+        get relevant values for finding a variant row within a MT
+        :return:
+        """
+        return hl.Locus(contig=f'chr{self.chr}', position=self.pos), [
+            self.ref,
+            self.alt,
+        ]
+
+    def __repr__(self):
+        return f'{self.chr}:{self.pos}_{self.ref}>{self.alt} - {self.confidence}'
+
+    def __eq__(self, other):
+        return (
+            self.chr == other.chr
+            and self.pos == other.pos
+            and self.ref == other.ref
+            and self.alt == other.alt
+        )
+
+    def __hash__(self):
+        """
+        hash function
+        :return:
+        """
+        return hash(
+            repr(self) + ''.join(list(map(lambda x: x.value, sorted(self.confidence))))
+        )
+
+
+CommonDict = Dict[str, List[CommonFormatResult]]
+
+
+def common_format_from_results(results_dict: Dict[str, Any]) -> CommonDict:
+    """
+    Parses the JSON
+
+    :param results_dict:
+    :return:
+    """
+    sample_dict: CommonDict = defaultdict(list)
+
+    # collect all per-sample results into a separate index
+    for sample, variants in results_dict.items():
+
+        for var in variants.values():
+            coords = var['var_data']['coords']
+            sample_dict[sample].append(
+                CommonFormatResult(
+                    coords['chrom'],
+                    int(coords['pos']),
+                    coords['ref'],
+                    coords['alt'],
+                    [Confidence.EXPECTED],
```

Comment:
It is not clear to me why you are using Expected as a default value here - but I am guessing it will become clear with the rest of the implementation.

Reply:
The thinking here is just that 'expected' is the highest confidence of the 3 options from seqr, and this is the default from the AIP side as 'we already got the result, so we're as confident as can be that we could do it again'. This could just as easily be an empty list...